mypy/newsemanal/semanal.py

```diff
@@ -284,6 +284,30 @@ def add_plugin_dependency(self, trigger: str, target: Optional[str] = None) -> N
             target = self.scope.current_target()
         self.cur_mod_node.plugin_deps.setdefault(trigger, set()).add(target)
 
+    def add_implicit_module_attrs(self, file_node: MypyFile) -> None:
+        """Manually add implicit definitions of module '__name__' etc."""
+        for name, t in implicit_module_attrs.items():
+            # unicode docstrings should be accepted in Python 2
+            if name == '__doc__':
+                if self.options.python_version >= (3, 0):
+                    typ = UnboundType('__builtins__.str')  # type: Type
+                else:
+                    typ = UnionType([UnboundType('__builtins__.str'),
+                                     UnboundType('__builtins__.unicode')])
+            else:
+                assert t is not None, 'type should be specified for {}'.format(name)
+                typ = UnboundType(t)
+
+            an_type = self.anal_type(typ)
+            if an_type:
+                var = Var(name, an_type)
+                var._fullname = self.qualified_name(name)
+                var.is_ready = True
```

Comment:
Is this necessary? Looks like we don't set this in the old semantic analyzer.

Reply:
We do this here https://github.com/python/mypy/blob/master/mypy/semanal.py#L301 the logic used to be split between passes. I am not sure if it is actually useful, but probably it is safer to keep it for 1:1 correspondence with old analyzer.

Reply:
Makes sense!